### DISCUSSION TITLE
Solid Start is unusable from the get-go

### DISCUSSION THREAD

**Discussion by JaielZeus** [solidjs/solid-start#912] [General]
Hello, 

I've come into Solid Start with the mindset it is not suitable for production yet, but I didn't think it would be THAT bad.

I was setting up a project to get into Solid Start with 
```
npm init solid/@latest
```
and opted into bare-bones. 

I added ``sass``, changed ``root.css`` to ``root.scss`` and was happy to see it works so far. Then I added a route/folder ``about`` into ``routes`` folder with the content of the root ``index.tsx`` and my happy demeanor went away.

It didn't work, I couldn't tell why. I tried changing names, contents it didn't work for the first 9 times. But on the 10th it worked finally.

Ok maybe a hiccup. So I copy paste ``about`` folder and renamed it ``about2`` AND AGAIN only the root route ``/`` was working, the other 2 routes did not work. AND AGAIN after 9 times trying something, restarting dev server, renaming files, folders and contents NOTHING but on the 10th it suddenly worked. REPEATING THE SAME PROCESS IT STARTS ALL OVER AGAIN.

Well I thought ok, maybe it will work sometimes on dev and sometimes it just doesn't. Maybe in production it works. So I went and did ``npm run build`` and it is stuck in a forever loop seemingly doing:

```
solid-start building client...
solid-start rendering index.html..
```

It is simply unusable to even develop or tinker with Solid Start like that. I haven't been able to go past trying out to set Routes and I am giving up on Solid Start now for the forseeable future. I really love solid for being so performant and all so I will continue to use that. But there is a dire need for Solid Start to be a solid implementation of a framework. Maybe I did something wrong idk, maybe using sass made something or just using the barebones template. It is really frustrating I did expect minor inconveniences but not such blatant ones!

**Comment by boehs** [comment]
It's hard to say what the specific issue is, but I've experienced non of this after about 100k total lines in solid start....

**Reply by JaielZeus** [reply]
I have selected CSR if that says something, maybe there is an issue

**Comment by JaielZeus** [comment]
Related issue is now #913 for the building process. Turns out in non-SSR the paths cannot contain a space in them, like it is possible on Windows. The SSR works fine and builds fine.

Furthermore my issue with HMR not updating correctly may be linked to this issue as well. I found that after I added a new route it would not update correctly, sometimes it updated when I changed the content multiple times, sometimes it did it randomly by just reloading the page.
